bokeh/server/auth_provider.py

```diff
@@ -89,7 +89,7 @@ def endpoints(self) -> List[Tuple[str, Type[RequestHandler]]]:
         return endpoints
 
     @property
-    def get_login_url(self) -> Callable[[HTTPServerRequest], str] | str | None:
+    def get_login_url(self) -> Callable[[HTTPServerRequest], str] | None:
```

Comment:
I'm not sure where any of the removed types in this module came from. Their presence contradicts the documentation, and I also can't find codepaths that correlate to them.

Reply:
This code was particularly bad to type, so those may have been a mistake, or a leftover from earlier iterations.